### DISCUSSION TITLE
Redundancy in database

### DISCUSSION THREAD

**Discussion by HolonProduction** [FossifyOrg/General-Discussion#575] [General]
Trying to implement support for adding the same song to a playlist multiple times I kind of hit a dead end, since the way this relation is modeled in the database makes it impossible to implement.

Instead of an m:n relation / crossref table, tracks get duplicated for each playlist they are added to and the combination of playlist and track idea is the primary key.

Is there a reason to not model this with m:n or is it just technical debt?

Also I noticed redundancy for other stuff like album names, artist names and some other fields the are stored in the track and the corresponding entity as well. Any reason for this?

**Comment by naveensingh** [comment]
>  Is there a reason to not model this with m:n or is it just technical debt?

Latter.  Efficiency **was** not a priority back then.
